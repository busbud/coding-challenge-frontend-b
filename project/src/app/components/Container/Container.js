import React from 'react'

import classes from './Container.module.sass'
import Header from '../Header/Header'

/**
 * container, include header,
 * navbar and content
 * without sidebar and footer
 */
const Container = ({ navbar, content }) => {
    return (
        <div className={classes.Container}>
            <Header />
            <div className={classes.Navbar}>
                {navbar}
            </div>
            <div className={classes.Content}>
                {content}
            </div>
        </div>
    )
}

export default Container