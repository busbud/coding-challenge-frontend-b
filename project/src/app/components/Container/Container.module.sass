@import '../../../shared/Styles/Colors.sass'

.Container
	min-height: 100vh
	background-color: $primary-color
	color: $text-color
	.Navbar, .Content
		padding: 0 10vw
		@media (max-width: 767px)
			padding: 0 5vw