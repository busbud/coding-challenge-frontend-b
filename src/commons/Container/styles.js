import styled from "styled-components";

import { BREAKPOINTS } from "utils/constants";

export const StyledContainer = styled.div`
  width: 100%;
  padding: 0 16px;

  @media screen and (min-width: ${BREAKPOINTS.tablet}) {
    width: 980px;
    padding: 0;
    margin: 0 auto;
  }
`;
