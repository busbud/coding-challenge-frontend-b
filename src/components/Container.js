import React from 'react'
import Styled from 'styled-components'

const Container = Styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1140px;
`
export default ({ children }) => <Container>{children}</Container>
