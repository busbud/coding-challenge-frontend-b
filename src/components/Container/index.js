import React from 'react'

import {
  StyledContainer
} from './styles'

const Container = (props) => <StyledContainer>{props.children}</StyledContainer>

export default Container