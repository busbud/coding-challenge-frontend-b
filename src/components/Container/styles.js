import styled from 'styled-components'

export const StyledContainer = styled.div`
  padding: 0 10px;
  margin: 0 auto;
  max-width: 620px;
  width: 100%;
`
