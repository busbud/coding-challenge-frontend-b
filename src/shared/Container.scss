.container {
    @include span(12);
    
    /** Large Desktop **/
    @media (min-width: $large-width) {
        @include span(8);
        @include push(2);
    }
}