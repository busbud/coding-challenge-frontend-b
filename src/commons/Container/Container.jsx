import React from "react";
import PropTypes from "prop-types";

import { StyledContainer } from "./styles";

export const Container = ({ children }) => (
  <StyledContainer>{children}</StyledContainer>
);

Container.propTypes = {
  /**
   * Any React of DOM children node element.
   */
  children: PropTypes.any.isRequired,
};

export default Container;
