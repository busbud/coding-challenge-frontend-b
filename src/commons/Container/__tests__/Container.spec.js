import React from "react";
import { render } from "@testing-library/react";

import Container from "../";

describe("<Container />", () => {
  it("should render properly", () => {
    const { container } = render(<Container>content</Container>);

    expect(container).toMatchSnapshot();
  });
});
