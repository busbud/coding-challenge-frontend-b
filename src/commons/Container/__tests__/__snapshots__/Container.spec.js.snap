// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render properly 1`] = `
<div>
  <div
    class="sc-bdfBwQ kZkkxR"
  >
    content
  </div>
</div>
`;
